enterprise/tpa_pipeline.py

```diff
@@ -20,24 +20,41 @@
 except ImportError:
     Registry = None
 
+try:
+    from third_party_auth.pipeline import get as get_pipeline_partial
+except ImportError:
+    get_pipeline_partial = None
+
+
+def verify_third_party_auth_dependencies():
+    """
+    Ensure that all necessary third_party_auth dependencies are present.
+    """
+    third_party_dependencies = (
+        Registry,
+        get_pipeline_partial,
+    )
+
+    if any(third_party_dependency is None for third_party_dependency in third_party_dependencies):
+        raise NotConnectedToEdX(
+            _("This package must be installed in an EdX environment to look up third-party auth dependencies.")
+        )
+
 
 def get_enterprise_customer_for_request(request):
     """
     Get the EnterpriseCustomer associated with a particular request.
     """
-    pipeline = request.session.get('partial_pipeline')
+    verify_third_party_auth_dependencies()
+    pipeline = get_pipeline_partial(request)
     return get_ec_for_running_pipeline(pipeline)
 
 
 def get_ec_for_running_pipeline(pipeline):
```

Comment:
The name of this method should probably be "get_enterprise_customer_for_running_pipeline" although that may result in a pylint violation due to length